toolchain/check/testdata/interop/cpp/function/struct.carbon

```diff
@@ -44,6 +44,78 @@ fn F() {
   Cpp.foo({});
 }
 
+// --- fail_todo_import_decl_value_param_type_previously_imported.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "decl_value_param_type.h";
+
+fn F() {
+  // CHECK:STDERR: fail_todo_import_decl_value_param_type_previously_imported.carbon:[[@LINE+11]]:10: error: semantics TODO: `Unsupported: Record declarations without a definition` [SemanticsTodo]
+  // CHECK:STDERR:   let s: Cpp.S;
+  // CHECK:STDERR:          ^~~~~
+  // CHECK:STDERR: fail_todo_import_decl_value_param_type_previously_imported.carbon:[[@LINE+8]]:10: note: in `Cpp` name lookup for `S` [InCppNameLookup]
+  // CHECK:STDERR:   let s: Cpp.S;
+  // CHECK:STDERR:          ^~~~~
+  // CHECK:STDERR:
+  // CHECK:STDERR: fail_todo_import_decl_value_param_type_previously_imported.carbon:[[@LINE+4]]:15: error: expected `=`; `let` declaration must have an initializer [ExpectedInitializerAfterLet]
+  // CHECK:STDERR:   let s: Cpp.S;
+  // CHECK:STDERR:               ^
+  // CHECK:STDERR:
+  let s: Cpp.S;
+  // CHECK:STDERR: fail_todo_import_decl_value_param_type_previously_imported.carbon:[[@LINE+7]]:3: error: semantics TODO: `Unsupported: parameter type: struct S` [SemanticsTodo]
+  // CHECK:STDERR:   Cpp.foo(s);
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR: fail_todo_import_decl_value_param_type_previously_imported.carbon:[[@LINE+4]]:3: note: in `Cpp` name lookup for `foo` [InCppNameLookup]
+  // CHECK:STDERR:   Cpp.foo(s);
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR:
+  Cpp.foo(s);
+}
+
+// ============================================================================
+// Forward-declared struct as parameter type imported twice
+// ============================================================================
+
+// --- double_decl_value_param_type.h
+
+struct S;
+
+auto foo1(S) -> void;
+auto foo2(S) -> void;
```

Comment:
Nothing about these tests seems to be intentionally causing an ErrorInst to be imported - perhaps they cause one now due to lack of support, but they won't indefinitely right? Could we produce a test that will always intentionally produce an ErrorInst on import? Or is that impossible because it would have to produce a clang error and then we wouldn't import it?

Reply:
Yes, not sure, and yes.
While we currently lack any support for declared but not defined structs, I'm not sure whether we plan to support calling a C++ function that takes declared but not defined structs by value.
It's possible the code would change to work differently by that time.
Given that these are integration tests and not unit tests, I'm not sure we should focus on how we import `ErrorInst`, as this is tied to the implementation, but rather make sure we cover the use cases to make sure behavior is as expected. These use cases are the most concise use cases to I found to trigger those crashes, to demonstrate the fix and make sure we won't have regressions for these use cases.